sub-packages/bionemo-amplify/src/bionemo/amplify/train_amplify.py

```diff
@@ -0,0 +1,337 @@
+        val_check_interval=val_check_interval,
+        log_every_n_steps=log_every_n_steps,
+        num_nodes=num_nodes,
+        callbacks=callbacks,
+        plugins=nl.MegatronMixedPrecision(
+            precision=precision,
+            params_dtype=get_autocast_dtype(precision),
+            pipeline_dtype=get_autocast_dtype(precision),
+            grad_reduce_in_fp32=grad_reduce_in_fp32,
+            autocast_enabled=False,
+        ),
+    )
+
+    tokenizer = BioNeMoAMPLIFYTokenizer()
+
+    # Initialize the data module.
+    data = AMPLIFYDataModule(
+        train_hf_dataset=hf_load_dataset("chandar-lab/UR100P", split="train"),  # type: ignore
+        valid_hf_dataset=hf_load_dataset("chandar-lab/UR100P", split="test"),  # type: ignore
+        global_batch_size=global_batch_size,
+        micro_batch_size=micro_batch_size,
+        min_seq_length=min_seq_length,
+        max_seq_length=max_seq_length,
+        num_workers=num_dataset_workers,
+        random_mask_strategy=random_mask_strategy,
+        tokenizer=tokenizer,
+    )
+
+    # Configure the model
+    train_metric = None
+    is_model_parallel = tensor_model_parallel_size * pipeline_model_parallel_size > 1
+    if is_model_parallel:
+        valid_metric = None  # metric logging under model parallelism is not supported yet
+    else:
+        valid_metric = TorchmetricsConfig(
+            class_path="text.Perplexity",
+            task="pretraining",
+            kwargs={"ignore_index": MLM_LOSS_IGNORE_INDEX},
+            metric_name="val_ppl",
+        )
+
+    amplify_config = AMPLIFYConfig(
+        seq_length=max_seq_length,
+        num_layers=num_layers,
+        hidden_size=hidden_size,
+        num_attention_heads=num_attention_heads,
+        ffn_hidden_size=ffn_hidden_size,
+        params_dtype=get_autocast_dtype(precision),
+        pipeline_dtype=get_autocast_dtype(precision),
+        autocast_dtype=get_autocast_dtype(precision),  # setting this speeds things up a lot
+        biobert_spec_option=biobert_spec_option,
+        nemo1_ckpt_path=str(nemo1_init_path) if nemo1_init_path is not None else None,
+        # handle checkpoint resumption here rather than auto-resume so this supports fine-tuning capabilities
+        initial_ckpt_path=str(restore_from_checkpoint_path) if restore_from_checkpoint_path is not None else None,
+        variable_seq_lengths=min_seq_length != max_seq_length,
+        train_metric=train_metric,
+        valid_metric=valid_metric,
+    )
+
+    model = biobert_lightning_module(
+        amplify_config,
+        tokenizer=tokenizer,
+        optimizer=MegatronOptimizerModule(
+            config=OptimizerConfig(
+                lr=lr,
+                optimizer="adam",  # fused_adam not supported
+                use_distributed_optimizer=False,
+                weight_decay=0.01,
+                adam_beta1=0.9,
+                adam_beta2=0.95,
+                clip_grad=1.0,
```

Comment:
These are so important to tune. I'd recommend pulling these out as options with these defaults to main.

Reply:
easy enough to do later if/when we need to change them. If AMGEN doesn't tune them i don't think we really need to either. Ideally we could do this as part of a config refactor, rather than having uncountably infinite CLI options